### PR TITLE
feat(file-uploader): adding reset and get files methods

### PR DESCRIPTION
**Pull request by Vignesh-Manogar-E3433**
https://github.com/freshworks/crayons/issues/716

**Description of the enhancement:**
Reset Method: add a reset method to make the component go back to the initial state.
getFiles: add a method to get the locally uploaded files from the component when part of a form.

There have been request from the community for the same.
https://community.freshworks.dev/t/file-uploader-how-to-reset/7282
https://community.freshworks.dev/t/how-to-use-fw-file-uploader/6881

We should be able to achieve something like,

```
const fileUploader = useRef(null);
const updateState = async (event) => {
  event.preventDefault();
  const files = await fileUploader.getFiles();
  console.log(files);
  fileUploader.reset();
};
```

```
<form onSubmit={(event) => updateState(event)}>
  <FwFileUploader
    name="sample-2"
    id="file-uploader-2"
    text="Upload PNG"
    accept=".png"
    ref={fileUploader}
  >
  </FwFileUploader>
  <FwButton type="submit">Upload</FwButton>
</form>
```

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My commits have standard messages as mentioned in [Contributing Guidelines](./../blob/next/CONTRIBUTING.md)

### CODE REVIEW

**Review by arvindanta — CHANGES_REQUESTED** [review]
The commit type is `test` but we have added some new props. Can you please change the commit type to `feat`

File: packages/crayons-core/src/components/file-uploader/readme.md [hunk 1]
```diff
@@ -86,6 +90,498 @@ fw-file-uploader can be used to upload files to a server.
...
+            hint="File size must be within 5MB"
+            max-file-size="5"
+            accept=".png"
+            action-u-r-l="/no-api"
+            onFwFilesUploaded={(event) => filesUploaded(event)}
+            onFwFileReuploaded={(event) => fileReuploaded(event)}
+            ref={fileUploader}
+          >
+          </FwFileUploader>
+          <FwButton file-uploader-id="file-uploader-3"></FwButton>
```
[the remark this answers is not in the document]
Comment (Vignesh-Manogar-E3433) [reply packages/crayons-core/src/components/file-uploader/readme.md]: Done.

File: packages/crayons-core/src/components/file-uploader/readme.md [hunk 2]
```diff
@@ -86,6 +90,498 @@ fw-file-uploader can be used to upload files to a server.
...
+    fileUploader2.reset(); // reset the form to initial state
+  });
+```
+
+</code-block>
+
+<code-block title="React">
+
+```jsx
+  import React from "react";
```
[the remark this answers is not in the document]
Comment (Vignesh-Manogar-E3433) [reply packages/crayons-core/src/components/file-uploader/readme.md]: Done.

**Comment by Vignesh-Manogar-E3433** [comment]
@arvindanta , Thanks for pointing out on the commit message. Requested changes are done.

**Review by arvindanta — APPROVED** [review]

**Pull request merged**